airsonic-main/src/main/java/org/airsonic/player/domain/PlayQueue.java

```diff
@@ -399,9 +409,6 @@ public synchronized Status getStatus() {
      */
     public synchronized void setStatus(Status status) {
         this.status = status;
-        if (index == -1) {
-            index = Math.max(0, Math.min(index, size() - 1));
-        }
     }
```

Comment:
I'm not quite sure if removal of this line could cause any side effects. I had to remove this to fix the following circumstances:

    Play a song
    Clear the playlist
    Add an album (just click on the + on the cover or use the "Add All" link, don't press play)
    First song got marked as current, even though it's not the one which is currently played by the controls, the controls were still playing the last song which isn't in the playlist anymore after clearing it

The method addFilesAt is invoking setStatus which was always resetting the index to 0 even though it should be -1.